src/back/backend_handler.rs

```diff
@@ -53,12 +53,19 @@ impl BackEndHandler {
     /// the request.
     ///
     /// # Errors
+    /// - if the provider ID can not perform the type of operation, returns
+    /// `ResponseStatus::PsaErrorNotSupported`
     /// - if the provider ID does not match, returns `ResponseStatus::WrongProviderID`
     /// - if the content type does not match, returns `ResponseStatus::ContentTypeNotSupported`
     /// - if the accept type does not match, returns `ResponseStatus::AcceptTypeNotSupported`
     pub fn is_capable(&self, request: &Request) -> Result<()> {
         let header = &request.header;
 
+        if (self.provider_id == ProviderID::Core) ^ header.opcode.is_core() {
```

Comment:
```suggestion
        if (self.provider_id == ProviderID::Core) != header.opcode.is_core() {
```

Seems easier to read/understand

Reply:
I think though with `!=` it will not go into the `if` in the following counter-example:
* `self.provider_id` is `Provider::Tpm`
* `header.opcode` is `Opcode::Ping`

Reply:
I will update, your version is definitely better